tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -803,3 +808,421 @@ def test_csrlsvqr(self, dtype):
+                                 contiguous_check=False)
+    def test_fiedler_small_8(self, xp, sp):
+        """Check the dense workaround path for small matrices.
+        """
+        # This triggers the dense path because 8 < 2*5.
+        return self._check_fiedler(8, 2, xp, sp)
+
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp',
+                                 contiguous_check=False)
+    def test_fiedler_large_12(self, xp, sp):
+        """Check the dense workaround path avoided for non-small matrices.
+        """
+        # This does not trigger the dense path, because 2*5 <= 12.
+        return self._check_fiedler(12, 2, xp, sp)
+
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp',
+                                 contiguous_check=False)
+    def test_eigs_consistency(self, xp, sp):
+        """Check eigs vs. lobpcg consistency.
+        """
+        # The n=5 case tests the alternative small matrix code path
+        param_arr = [(20, 1e-3), (5, 1e-8)]
+        eval_list = []
+        for n, _atol in param_arr:
+            vals = xp.arange(1, n + 1, dtype=xp.float64)
+            A = sp.spdiags(vals, 0, n, n)
+            X = testing.shaped_random((n, 2), xp=xp, seed=345678)
+            lvals, lvecs = sp.linalg.lobpcg(A, X, largest=True, maxiter=100)
+            eval_list.append(lvals)
+            eval_list.append(self._eigen_vec_transform(lvecs, xp))
+            vals, _ = sp.linalg.eigsh(A, k=2)
+            self._check_eigen(A, lvals, lvecs, xp, sp, atol=_atol, rtol=0)
+            cupy.testing.assert_allclose(xp.sort(vals), xp.sort(lvals),
+                                         atol=1e-14)
+        return eval_list
+
+    def _verbosity_helper(self, xp, sp):
+        """Helper to capture the verbose output from stdout
+        """
+        A, B = self._elastic_rod(100, xp)
+        n = A.shape[0]
+        m = 20
+        V = testing.shaped_random((n, m), xp=numpy, seed=0)
+        X = scipy.linalg.orth(V)
+        saved_stdout = io.StringIO()
+        with contextlib.redirect_stdout(saved_stdout):
+            _, _ = sp.linalg.lobpcg(A, xp.asarray(X), B=B, tol=1e-5,
+                                    maxiter=30, largest=False,
+                                    verbosityLevel=9)
+        output = saved_stdout.getvalue().strip()
+        return output
+
+    def test_verbosity(self):
+        """Check that nonzero verbosity level code runs
+           and is identical to scipy's output format.
+        """
+        stdout_cupy = self._verbosity_helper(cupy, cupyx.scipy.sparse)
+        stdout_numpy = self._verbosity_helper(numpy, scipy.sparse)
+        # getting rid of the numbers and whitespaces, we care only about
+        # format of printed output.
+        # also, due to the fact that there are unpredictable (but minor)
+        # differences in decimal digits between scipy and cupy verbose output
+        stdout_cupy = re.sub(r'[0-9\.\-\+ \t]+', '', stdout_cupy)
```

Comment:
How about using `r'[-+]?\d+\.\d+'` as the floating number pattern?

Reply:
Could you replace number strings with non-empty string (e.g., `'{number}'`) for robustness?